ml-agents/tests/yamato/training_int_tests.py

```diff
@@ -0,0 +1,84 @@
+import os
+import sys
+import subprocess
+
+from .yamato_utils import (
+    get_base_path,
+    run_standalone_build,
+    init_venv,
+    override_config_file,
+    get_rewards_stats,
+)
+
+
+def main():
+    nn_file_expected = "./models/ppo/3DBall.nn"
+    if os.path.exists(nn_file_expected):
+        # Should never happen - make sure nothing leftover from an old test.
+        print("Artifacts from previous build found!")
+        sys.exit(1)
+
+    base_path = get_base_path()
+    print(f"Running in base path {base_path}")
+
+    build_returncode = run_standalone_build(base_path)
+    if build_returncode != 0:
+        print("Standalone build FAILED!")
+        sys.exit(build_returncode)
+
+    init_venv()
+
+    # Copy the default training config but override the max_steps parameter
+    override_config_file("config/trainer_config.yaml", "override.yaml", max_steps=100)
+
+    # TODO pass scene name and exe destination to build
+    # TODO make sure we fail if the exe isn't found - see MLA-559
+    mla_learn_cmd = "mlagents-learn override.yaml --train --env=UnitySDK/testPlayer --no-graphics --env-args -logFile -"  # noqa
+    res = subprocess.run(f"source venv/bin/activate; {mla_learn_cmd}", shell=True)
+
+    if res.returncode != 0 or not os.path.exists(nn_file_expected):
+        print("mlagents-learn run FAILED!")
+        sys.exit(1)
+
+    full_training = False
+    if full_training:
+        # Run for longer and make sure we get a reasonable reward
+        # 3DBall should reach its max score (100.0) within about 200K agent steps,
+        # which takes about 2 minutes on a laptop (but longer on a bokken VM)
+        # TODO try larger flavors with multiple envs
+        override_config_file(
+            "config/trainer_config.yaml", "override.yaml", max_steps=200000
+        )
+        target_max_reward = 100.0
```

Comment:
There are a lot of hard coded numbers in the code, should they be gathered somewhere?

Reply:
I'll remove this block for now, and we can figure out how to store these later.